src/index.js

```diff
@@ -49,73 +50,117 @@ function createCommand(queryName, implementationName) {
+        Selector: getSelector(),
+        'Applied To': getContainer(
+          options.container || prevSubject || win.document,
+        )
       }
 
-      Cypress.log({
-        $el: inputArr,
-        name: queryName,
-        message: inputArr,
-        consoleProps: () => consoleProps,
-      })
+      if (options.log) {
+        options._log = Cypress.log({
+          name: queryName,
+          message: inputArr,
+          consoleProps: () => consoleProps,
+        })
+      }
 
-      return cy
-        .window({log: false})
-        .then({timeout: waitOptions.timeout + 100}, thenArgs => {
-          const getValue = () => {
-            const value = commandImpl(thenArgs.document)
-            const result = Cypress.$(value)
-
-            // Overriding the selector of the jquery object because it's displayed in the long message of .should('exist') failure message
-            // Hopefully it makes it clearer, because I find the normal response of "Expected to find element '', but never found it" confusing
-            result.selector = `${queryName}(${queryArgument(args)})`
-
-            if (result.length > 0) {
-              consoleProps.yielded = result.toArray()
-            }
-
-            return result
-          }
-
-          const resolveValue = () => {
-            // retry calling "getValue" until following assertions pass or this command times out
-            return Cypress.Promise.try(getValue).then(value => {
-              return cy.verifyUpcomingAssertions(value, waitOptions, {
-                onRetry: resolveValue,
-              })
-            })
-          }
+      const getValue = () => {
+        const value = commandImpl(win.document)
+
+        const result = Cypress.$(value)
+        if (value && options._log) {
+          options._log.set('$el', result)
+        }
 
-          if (queryRegex.test(queryName)) {
-            // For get* queries, do not retry
-            return getValue()
-          }
+        // Overriding the selector of the jquery object because it's displayed in the long message of .should('exist') failure message
+        // Hopefully it makes it clearer, because I find the normal response of "Expected to find element '', but never found it" confusing
+        result.selector = getSelector()
 
-          return resolveValue().then(subject => {
-            // Remove the error that occurred because it is irrelevant now
-            if (consoleProps.error) {
-              delete consoleProps.error
-            }
+        consoleProps.elements = result.length
+        if (result.length === 1) {
+          consoleProps.yielded = result.toArray()[0];
+        } else if (result.length > 0) {
+          consoleProps.yielded = result.toArray();
+        }
 
-            return subject
-          })
+        if (result.length > 1 && !/all/i.test(queryName)) {
+          // Is this useful?
+          throw Error(`Found multiple elements with the text: ${queryArgument(args)}`)
+        }
```

Comment:
This line is part of the question asking if we should throw errors for the `*By` queries. It is what the other `@testing-library` libraries do

Reply:
Let's change it to case-sensitive (just in case): `!/All/.test(queryName)`.

I think that we should be consistent with the other libraries 👍